diff -urN linux-2.6.9/arch/i386/Kconfig linux-2.6.9-qemu_100hz/arch/i386/Kconfig
--- linux-2.6.9/arch/i386/Kconfig	2004-10-18 23:53:22.000000000 +0200
+++ linux-2.6.9-qemu_100hz/arch/i386/Kconfig	2004-11-01 20:02:47.703053072 +0100
@@ -337,6 +337,13 @@
 
 endif
 
+config QEMU_100_HZ
+       bool "Build 100 HZ kernel to run under QEMU"
+       depends on EXPERIMENTAL
+       help
+         Select this if you want to boot the kernel inside qemu,
+         correct timing requires 100 HZ.  Say N.
+
 #
 # Define implied options from the CPU selection here
 #
diff -urN linux-2.6.9/include/asm-i386/param.h linux-2.6.9-qemu_100hz/include/asm-i386/param.h
--- linux-2.6.9/include/asm-i386/param.h	2004-10-18 23:53:24.000000000 +0200
+++ linux-2.6.9-qemu_100hz/include/asm-i386/param.h	2004-11-01 20:04:23.654466240 +0100
@@ -2,7 +2,12 @@
 #define _ASMi386_PARAM_H
 
 #ifdef __KERNEL__
-# define HZ		1000		/* Internal kernel timer frequency */
+# include <linux/config.h>
+# ifdef CONFIG_QEMU_100_HZ
+#  define HZ            100
+# else
+#  define HZ            1000            /* Internal kernel timer frequency */
+# endif
 # define USER_HZ	100		/* .. some user interfaces are in "ticks" */
 # define CLOCKS_PER_SEC		(USER_HZ)	/* like times() */
 #endif
